src/Microsoft.Fx.Portability/DataExtensions.cs

```diff
@@ -33,13 +33,13 @@ public static class DataExtensions
 
         public static byte[] Serialize<T>(this T data)
         {
-            var str = JsonConvert.SerializeObject(data, Formatting.Indented, JsonSettings);
-
             using (var outputStream = new MemoryStream())
-            using (var writer = new StreamWriter(outputStream))
             {
-                writer.Write(str);
-                writer.Flush();
+                using (var writer = new StreamWriter(outputStream))
```

Comment:
I added [`SerializeAndCompress<T>`(this T data)](https://github.com/Microsoft/dotnet-apiport/pull/403/files#diff-e913c19f08e1e43f4dc9847354ec853bR50).  Does this work?